### PR TITLE
improve Package search sorting

### PR DESCRIPTION
**Pull request by jderochervlk**
- increase number of included packages from 250 to 750 (currently 512)
- filter out packages with a low maintenance score (< 0.3) and add in a filter to show "outdated" packages
- add query params to npm registry api call to adjust weights to favor maintenance and quality
- sort by weight in the UI
- filter out packages with "reason" in the title to keep the focus on ReScript specific packages

image

| Before | After |
| -- | -- |
| image | image |
|image | image|

### CODE REVIEW

File: src/Packages.res [hunk 1]
```diff
@@ -538,9 +566,46 @@ let getStaticProps: Next.GetStaticProps.revalidate<props, unit> = async _ctx =>
...
+  ))
+
+  let unmaintained = []
+
+  let pkges =
+    parsePkgs(data1)
+    ->Js.Array2.concat(parsePkgs(data2))
+    ->Js.Array2.concat(parsePkgs(data3))
+    ->Js.Array2.filter(pkg => {
+      if [/* Allow list of names */]->Js.Array2.includes(pkg.name) {
```
Comment (fhammerschmidt) [inline src/Packages.res:597]: You could do a 

```res
let allowList = []
```

somehwere at the beginning of the file already then you don't need the comment here.

**Review by fhammerschmidt — CHANGES_REQUESTED** [review]
very nice work

File: src/Packages.res (hunk 1 above)
Comment (jderochervlk) [reply src/Packages.res]: Done.

File: src/Packages.res [hunk 2]
```diff
@@ -23,21 +23,28 @@ type npmPackage = {
   description: string,
   repositoryHref: Js.Null.t<string>,
   npmHref: string,
+  searchScore: float,
+  maintenanceScore: float,
+  score: {"final": float, "detail": {"quality": float, "popularity": float, "maintenance": float}},
```
Comment (fhammerschmidt) [inline src/Packages.res:28]: Is there any reason why you prefer an object here? Otherwise I suggest to make it a record.

Comment (jderochervlk) [reply src/Packages.res]: I was just copying from the raw source type to test out an idea, but I didn't end up using it so I went ahead and removed this from the type.

File: src/Packages.res [hunk 3]
```diff
@@ -116,35 +124,41 @@ module Resource = {
...
+      (npms, resources, outdated)
     })
 
     let filteredNpm = applyNpmSearch(allNpms, pattern)->Belt.Array.map(m => Npm(m["item"]))
     let filteredUrls = applyUrlResourceSearch(allUrls, pattern)->Belt.Array.map(m => Url(m["item"]))
+    let filteredOutdated =
+      applyNpmSearch(allOutDated, pattern)->Belt.Array.map(m => Outdated(m["item"]))
 
-    Belt.Array.concat(filteredNpm, filteredUrls)
+    Belt.Array.concat(filteredNpm, filteredUrls)->Belt.Array.concat(filteredOutdated)
```
Comment (fhammerschmidt) [inline src/Packages.res:146]: there is also Belt.Array.concatMany([])

Comment (jderochervlk) [reply src/Packages.res]: Fixed.

**Review by fhammerschmidt — APPROVED** [review]

**Pull request merged**